### PR TITLE
extend ProtectedError and RestrictedError stubs

### PR DESCRIPTION
**Pull request by timnyborg**
types for their init methods and *_objects properties

Solves `"ProtectedError" has no attribute "protected_objects"` and similar false positives

### CODE REVIEW

File: django-stubs/db/models/deletion.pyi [hunk 1]
```diff
@@ -14,8 +14,13 @@ def RESTRICT(collector, field, sub_objs, using): ...
 def SET(value: Any) -> Callable: ...
 def get_candidate_relations_to_delete(opts: Options) -> Iterable[Field]: ...
 
-class ProtectedError(IntegrityError): ...
-class RestrictedError(IntegrityError): ...
+class ProtectedError(IntegrityError):
+    protected_objects: set[Model]
```
Comment (sobolevn) [inline django-stubs/db/models/deletion.pyi]: We use `typing.Set`

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks!

**Comment by timnyborg** [comment]
Thanks, I've replaced with set with typing.Set

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks!

**Pull request merged**